### ISSUE TITLE
Embedded lowcost GPU

### ISSUE THREAD

**Issue by rebotnix** [ZHKKKe/MODNet#3]
Hi,

cool project. Would like to test if we can port it to an ARM64 with a 32GB embedded low power GPU system (NVIDIA AGX Xavier 32GB). This can be used for training and inferencing. A first test with the pretrained in Dec. 2020 would give us a first impression of how the performance is comparable with desktop GPU.

**Comment by ZHKKKe** [comment]
Hi, thank you for your good suggestion. See if we can achieve this goal after the code is released. I will **@** you when the inference code is available.   :)

**Comment by ZHKKKe** [comment]
@rebotnix 
Thanks for your patience. Our demos and pre-trained models are released.

**Comment by rebotnix** [comment]
Hi ZHKKKe,

would like to let you know that i was able to run the inference on NVIDIA NX and AGX. Really good job. 

Tested with one image yet, will try video and TRT acceleration soon. See my picture blew. 

![out](https://user-images.githubusercontent.com/566761/102149207-2d2a3e80-3e6e-11eb-9667-826d245f136a.png)

For Developers:
If you want to save the image instead of displaying it, just add 

`def combined_display(image, matte):
	  # calculate display resolution
	  w, h = image.width, image.height
	  rw, rh = 800, int(h * 800 / (3 * w))
	  
	  # obtain predicted foreground
	  image = np.asarray(image)
	  if len(image.shape) == 2:
	    image = image[:, :, None]
	  if image.shape[2] == 1:
	    image = np.repeat(image, 3, axis=2)
	  elif image.shape[2] == 4:
	    image = image[:, :, 0:3]
	  matte = np.repeat(np.asarray(matte)[:, :, None], 3, axis=2) / 255
	  foreground = image * matte + np.full(image.shape, 255) * (1 - matte)
	  
	  # combine image, foreground, and alpha into one line
	  combined = np.concatenate((image, foreground, matte * 255), axis=1)
	  combined = Image.fromarray(np.uint8(combined)).resize((rw, rh))
	  return combined`

Call it
`  input_folder = "images"
    output_folder = "output"

    image_names = os.listdir(input_folder)

    for image_name in image_names:
    	matte_name = image_name.split('.')[0] + '.png'
    	image = Image.open(os.path.join(input_folder, image_name))
    	matte = Image.open(os.path.join(output_folder, matte_name))
    	res = combined_display(image, matte)
    	res.save("out.png")`

Do not need colab or anything else then. Maybe that helps.

**Comment by ZHKKKe** [comment]
Great! Big thanks!
